src/metabase/sync/sync_metadata/fields.clj

```diff
@@ -68,20 +69,25 @@
      ;; `sync-instances`
      (sync-metadata/update-metadata! table db-metadata (fields.our-metadata/our-metadata table))))
 
-(mu/defn sync-fields-for-db!
-  "Sync the Fields in the Metabase application database for a specific `table`."
+(mu/defn sync-fields! :- [:map
+                          [:updated-fields ms/IntGreaterThanOrEqualToZero]
+                          [:total-fields   ms/IntGreaterThanOrEqualToZero]]
+  "Sync the Fields in the Metabase application database for all the Tables in a `database`."
   [database :- i/DatabaseInstance]
   (sync-util/with-error-handling (format "Error syncing Fields for Database ''%s''" (sync-util/name-for-logging database))
-    (let [schema-names    (sync-util/db->sync-schemas database)
-          fields-metadata (fetch-metadata/fields-metadata database :schema-names schema-names)]
+    (let [driver          (driver.u/database->driver database)
+          schemas?        (driver/database-supports? driver :schemas database)
+          fields-metadata (if schemas?
+                            (fetch-metadata/fields-metadata database :schema-names (sync-util/db->sync-schemas database))
+                            (fetch-metadata/fields-metadata database))]
```

Comment:
This change fixes a (hypothetical) bug if there was driver that doesn't support schemas that did implement `describe-fields`. `db->sync-schemas` assumes that the database supports schemas. This is the case for the only (officially supported) database that supports `describe-fields`, but it might not be in the future.